source/extensions/extensions_build_config.bzl

```diff
@@ -199,8 +199,8 @@ EXTENSIONS = {
 
 }
 
-# This can be used to extend the visibility rules for Envoy extensions
-# (//:extension_config and //:extension_library in //BUILD)
-# if downstream Envoy builds need to directly reference envoy extensions.
-ADDITIONAL_VISIBILITY = [
-  ]
+
+# These can be changed to ["//visibility:public"], for  downstream builds which
+# need to directly reference Envoy extensions.
+EXTENSION_CONFIG_VISIBILITY = ["//:extension_config"]
+EXTENSION_EXTENSION_PACKAGE_VISIBILITY = ["//:extension_library"]
```

Comment:
This seems reasonable. I can imagine a finer grained system, but this is fine as a starting point to limit the visibility in Envoy.